pkg/server/server_presentation_test.go

```diff
@@ -195,82 +331,82 @@ func TestPresentationAPI(t *testing.T) {
 				assert.Empty(tt, resp.Definitions)
 			})
 
-			t.Run("List definitions returns many definitions", func(tt *testing.T) {
-				s := test.ServiceStorage(tt)
-				pRouter, _ := setupPresentationRouter(tt, s)
-				def1 := createPresentationDefinition(tt, pRouter)
-				def2 := createPresentationDefinition(tt, pRouter)
+			tt.Run("List definitions returns many definitions", func(ttt *testing.T) {
+				s := test.ServiceStorage(ttt)
+				pRouter, _ := setupPresentationRouter(ttt, s)
+				def1 := createPresentationDefinition(ttt, pRouter)
+				def2 := createPresentationDefinition(ttt, pRouter)
 
 				req := httptest.NewRequest(http.MethodGet, "https://ssi-service.com/v1/presentations/definitions", nil)
 				w := httptest.NewRecorder()
 				c := newRequestContext(w, req)
 				pRouter.ListDefinitions(c)
-				assert.True(tt, util.Is2xxResponse(w.Code))
+				assert.True(ttt, util.Is2xxResponse(w.Code))
 
 				var resp router.ListDefinitionsResponse
-				assert.NoError(tt, json.NewDecoder(w.Body).Decode(&resp))
-				assert.Len(tt, resp.Definitions, 2)
-				assert.ElementsMatch(tt, resp.Definitions, []*exchange.PresentationDefinition{
+				assert.NoError(ttt, json.NewDecoder(w.Body).Decode(&resp))
+				assert.Len(ttt, resp.Definitions, 2)
+				assert.ElementsMatch(ttt, resp.Definitions, []*exchange.PresentationDefinition{
 					&def1.PresentationDefinition,
 					&def2.PresentationDefinition,
 				})
 			})
 
-			t.Run("Create returns error without input descriptors", func(tt *testing.T) {
-				s := test.ServiceStorage(tt)
-				pRouter, _ := setupPresentationRouter(tt, s)
+			tt.Run("Create returns error without input descriptors", func(ttt *testing.T) {
```

Comment:
the idea is that each test runner should be able to act independently, if we look at what's provided by `t *testing.T` we see that it's a test context responsible for the lifecycle of a test...so separating it out per running test has a few benefits:
- a child failing test won't result in a parent failing test (we want to see all failures individually)
- individual tests can be skipped / toggled by using `t.Skip()`
- individual tests can have their own cleanup functions using `t.Cleanup()`

probably more too. just better isolation/separation of concerns

separately -- I think it's a mistake to couple storage to these tests. I believe we should have a separate set of storage tests that do a kind of "steel thread" for the service's functionality instead of running _every test_ for _every storage_ provider